File: backend/src/main/java/com/school/saas/module/user/UserDTO.java
package com.school.saas.module.user;

import com.school.saas.common.Role;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserDTO {
    private UUID id;
    private UUID schoolId;
    private String email;
    private String firstName;
    private String lastName;
    private String phone;
    private Role role;
    private Boolean enabled;
    private LocalDateTime lastLoginAt;
}

File: mobile/lib/models/user_dto.dart
class UserDTO {
  final String id;
  final String schoolId;
  final String email;
  final String firstName;
  final String lastName;
  final String role;

  UserDTO({
    required this.id,
    required this.schoolId,
    required this.email,
    required this.firstName,
    required this.lastName,
    required this.role,
  });

  factory UserDTO.fromJson(Map<String, dynamic> json) {
    return UserDTO(
      id: json['id'],
      schoolId: json['schoolId'],
      email: json['email'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      role: json['role'],
    );
  }
}
